frontend/src/metabase/visualizations/lib/settings/graph.js

```diff
@@ -151,26 +151,38 @@ export const GRAPH_DATA_SETTINGS = {
     dashboard: false,
     useRawSeries: true,
   },
+  "graph.series_order_dimension": {
+    getValue: (_series, settings) => settings["graph.dimensions"][1],
+    readDependencies: ["graph.series_order"],
```

Comment:
Due to the way we generate the computed settings map, this property ensures that `graph.series_order` will be calculated *before* we calculate a value for `graph.series_order_dimension`. This means that if there is a value for `graph.series_order_dimension`, we will use it rather than use the computed value when evaluating `graph.series_order`. It's all very strange, but this is what allows us to do the comparison in `graph.series_order` to see if the dimension has changed or not.

Reply:
I agree that that would be useful. I'd also like to document some of these props in general, but I'm not sure where would be appropriate